src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c

```diff
@@ -21,69 +63,1180 @@ const ActorInit Boss_Fd2_InitVars = {
+    }
+    DECR(this->blinkTimer);
+
+    if (this->screamTimer != 0) {
+        phi_f0 = 6000.0f;
+        phi_f2 = 1300.0f;
+    } else {
+        phi_f0 = (this->varianceTimer & 0x10) ? 1000.0f : 0.0f;
+        phi_f2 = 700.0f;
+    }
+    Math_SmoothScaleMaxF(&this->jawOpening, phi_f0, 0.3f, phi_f2);
+
+    DECR(this->screamTimer);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D285C.s")
+void BossFd2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BossFd2* this = THIS;
+    s16 i;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D2930.s")
+    osSyncPrintf("FD2 move start \n");
+    this->disableAT = 0;
+    this->actor.flags &= ~0x400;
+    this->varianceTimer++;
+    this->unkTimer++;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D2A08.s")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/BossFd2_Init.s")
+    for (i = 0; i < 5; i++) {
+        if (this->timers[i] != 0) {
+            this->timers[i]--;
+        }
+    }
+    if (this->damageFlashTimer != 0) {
+        this->damageFlashTimer--;
+    }
+    if (this->invulnerabilityTimer != 0) {
+        this->invulnerabilityTimer--;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/BossFd2_Destroy.s")
+    if (this->cutsceneState == 0) {
+        if (this->invulnerabilityTimer == 0) {
+            BossFd2_CollisionCheck(this, globalCtx);
+        }
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        if (!this->disableAT) {
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D2BD0.s")
+    BossFd2_UpdateFace(this, globalCtx);
+    this->bodyTex1Scroll += 4.0f;
+    this->bodyTex1Rot = 120.0f;
+    this->bodyTex2Scroll += 3.0f;
+    this->bodyTex2Rot -= 2.0f;
+    if (this->actor.posRot2.pos.y < 90.0f) {
+        this->actor.flags &= ~1;
+    } else {
+        this->actor.flags |= 1;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D2CE0.s")
+s32 BossFd2_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                             Actor* thisx) {
+    BossFd2* this = THIS;
+    BossFd* bossFd = BOSSFD;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D308C.s")
+    if (limbIndex == 31) {
+        rot->y -= (f32)this->fireBreathAngle.y;
+        rot->z += (f32)this->fireBreathAngle.x;
+    }
+    if (limbIndex != 32) {
+        if ((limbIndex == 35) || (limbIndex == 36)) {
+            rot->z -= this->jawOpening * 0.1f;
+        }
+    } else {
+        rot->z += this->jawOpening;
+    }
```

Comment:
I guess
```
if (32) rot->z += this->jawOpening
else if (35 || 36) rot->z -= this->jawOpening * 0.1f
```
doesn't match? weird code

Reply:
I'm now pretty sure this is a short switch statement.